flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala

```diff
@@ -351,7 +351,7 @@ object ScalarOperators {
     else {
       s"""
         |${operand.code}
-        |boolean $resultTerm = false;
+        |boolean $resultTerm = ${operand.resultTerm} == null;
```

Comment:
This is not a bug but intended. The result type should be a primitive at this point. I remove it and all tests are still working correctly.